main/lsp/requests/completion.cc

```diff
@@ -124,6 +137,14 @@ SimilarMethodsByName similarMethodsForClass(const core::GlobalState &gs, core::S
             if (!memberSymbol.data(gs)->isMethod()) {
                 continue;
             }
+            if (hasAngleBrackets(memberName.data(gs)->shortName(gs))) {
+                // Gets rid of methods like `<test_foo bar>` generated by our DSL passes
```

Comment:
Do we mark these as being synthesized? Do we only want to remove synthesized methods when the contain angle brackets?

Reply:
I don't like the idea of treating "synthesized" methods specially. In my experience, each place you want to ask whether some symbol is "synthesized" the property you're really looking for is different from nearly all other places. It's better to check the actual property you'd like to be true in each place. For example:

- `isSynthesized` means: is under the MAX_SYNTHETIC_SYMBOLS limit, which has to do with the symbols we initialize global state with
- `isRewriterSynthesized` means: "vaguely came from a DSL pass" (but props are dsl-synthesized, and we definitely want to suggest those!)
- another example: `isSyntheticBlockArg`: whether or not the block arg actually existed in the MethodDef node in the tree, or whether we created it

so yes, i think that checking for `hasAngleBrackets` is the property i want here. I'd be fine checking for "isSyntacticallyValidMethodName" but i that is a harder property to validate, and I'd rather name my method what it's doing rather than what it aspires to do.

Reply:
That makes sense, thanks!